src/arbitrage.py

```diff
@@ -0,0 +1,39 @@
+class Query (object):
+
+    def __init__(self, limit=0, stock_names=None, api=None):
+        self.limit = limit
+        self.names = stock_names
+        self.api = api
+
+    def pull(self, stock):
+        """Connects to API to pull data for a given stock"""
+        return
+
+    def check(self, expiry_dates=None, diff_costs=None, underlying_cost=None):
+        """Checks to see if there exists an expiry date for which
+        the arbitrage formula for options holds"""
+        result = False
+        for expiry in expiry_dates:
+            diff_cost = diff_costs[expiry]
+            if diff_cost / underlying_cost > self.limit:
+                result = True
+        return result
+
+    def find_opportunities(self):
+        """Prints the stocks that satisfy the arbitrage opportunity requirements"""
+        for name in self.names:
+            data = self.pull(name)
+            expiry_dates = data.get_dates(name)
+            underlying_cost = data.get_underlying(name)
```

Comment:
Where would the underlying cost come from? Isn't it the same as spot price? Look at the design doc.

Reply:
fixed

Reply:
Why is there only one underlying cost here? You should be able to check for opportunities assuming you purchase on any given day in the past, not a specific one.

You should either change this to accept either a date and assume you want to check for arbitrage opportunities from that day until now, or accept time range and check for opportunities in that range.